File: testbed_ruby/test/associations/unidirectional_m_star_test.rb
require 'test/unit'
require 'src/student_a_l'
require 'src/mentor_a_l'
require 'src/program_a_l'

module CruiseTest
class UnidirectionalMStarTest < Test::Unit::TestCase

  def test_constructorTooFew
  	assert_raise RuntimeError do #"Unable to create MentorAL, must have at least 3 students"
      s = StudentAL.new(99)
      s2 = StudentAL.new(98)
      MentorAL.new("blah",[s,s2])
    end
  end

  def test_constructorRequiresMinimum
    s = StudentAL.new(99)
    s2 = StudentAL.new(98)
    s3 = StudentAL.new(97)
    
    m = MentorAL.new("blah",[s,s2,s3])
    assert_equal(3,m.number_of_students)
  end
  
  def test_addRemoveWithinLimits
    s = StudentAL.new(99)
    s2 = StudentAL.new(98)
    s3 = StudentAL.new(97)
    s4 = StudentAL.new(96)
    s5 = StudentAL.new(95)
    s6 = StudentAL.new(94)
    
    m = MentorAL.new("blah",[s,s2,s3])
    assert_equal(3,m.number_of_students)
    assert_equal(true,m.add_student(s4))
    assert_equal(true,m.add_student(s5))
    
    assert_equal(5,m.number_of_students)
    
    assert_equal(false,m.remove_student(s6))
    assert_equal(true,m.remove_student(s3))
    assert_equal(true,m.remove_student(s4))
    assert_equal(false,m.remove_student(s))
    assert_equal(3,m.number_of_students)
  end

  def test_deleteDoesNotChangeStudent
    s = StudentAL.new(99)
    s2 = StudentAL.new(98)
    s3 = StudentAL.new(97)

    p = ProgramAL.new
    s.set_program(p)
    
    m = MentorAL.new("blah",[s,s2,s3])
    m.delete
    assert_equal(0,m.number_of_students)
    assert_equal(p,s.get_program)
  end
end
end

File: testbed_php/test/associations/UnidirectionalMStarTest.php
<?php

class UnidirectionalMStarTest extends UnitTestCase
{

  public function test_constructorTooFew()
  {
  	$this->expectException(new Exception("Unable to create MentorAL, must have at least 3 students"));
    $s = new StudentAL(99);
    $s2 = new StudentAL(98);
    new MentorAL("blah",array($s,$s2));
  }
  
  

  public function test_constructorRequiresMinimum()
  {
    $s = new StudentAL(99);
    $s2 = new StudentAL(98);
    $s3 = new StudentAL(97);
    
    $m = new MentorAL("blah",array($s,$s2,$s3));
    $this->assertEqual(3,$m->numberOfStudents());
  }
  

  public function test_addRemoveWithinLimits()
  {
    $s = new StudentAL(99);
    $s2 = new StudentAL(98);
    $s3 = new StudentAL(97);
    $s4 = new StudentAL(96);
    $s5 = new StudentAL(95);
    $s6 = new StudentAL(94);
    
    $m = new MentorAL("blah",array($s,$s2,$s3));
    $this->assertEqual(3,$m->numberOfStudents());
    $this->assertEqual(true,$m->addStudent($s4));
    $this->assertEqual(true,$m->addStudent($s5));
    
    $this->assertEqual(5,$m->numberOfStudents());
    
    $this->assertEqual(false,$m->removeStudent($s6));
    $this->assertEqual(true,$m->removeStudent($s3));
    $this->assertEqual(true,$m->removeStudent($s4));
    $this->assertEqual(false,$m->removeStudent($s));
    $this->assertEqual(3,$m->numberOfStudents());
  }
  
  

  public function test_deleteDoesNotChangeStudent()
  {
    $s = new StudentAL(99);
    $s2 = new StudentAL(98);
    $s3 = new StudentAL(97);

    $p = new ProgramAL();
    $s->setProgram($p);
    
    $m = new MentorAL("blah",array($s,$s2,$s3));
    $m->delete();
    $this->assertEqual(0,$m->numberOfStudents());
    $this->assertEqual($p,$s->getProgram());
  }
}
